### DISCUSSION TITLE
Creating relation alongside model resource

### DISCUSSION THREAD

**Discussion by atapatel** [tailflow/laravel-orion#291] [Ideas]
First of all I would like to thank you tailflow team to create such a nice and wonder full package for Laravel Community. I just want to know How can I create resource with relational data.
For Example.
Product -> hasMany -> ProductVariant
-> I want to create single API point to create Product with ProductVariant then how can I do it?

**Comment by jeroenqui** [comment]
https://tailflow.github.io/laravel-orion-docs/v2.x/guide/query-parameters.html#including-relations

**Comment by atapatel** [comment]
@jeroenqui i guess that is for GET API.

**Comment by alexzarbn** [comment]
Hi @atapatel,

At this point, it is not possible to create both model and relation resource _in one query_. It is possible, however, to do that in two queries, but you would need both [model](https://tailflow.github.io/laravel-orion-docs/v2.x/guide/models.html#setting-up-controller) and [relation](https://tailflow.github.io/laravel-orion-docs/v2.x/guide/relationships.html#setting-up-controller) controllers in place.

Another option would be to utilize `afterStore` [hook](https://tailflow.github.io/laravel-orion-docs/v2.x/guide/hooks.html#on-model-controllers) on a model controller and write the storing logic for a relation there, but that kind of breaks the whole point of Orion, since you would need to write all that basic stuff all over again.

**Comment by atapatel** [comment]
@alexzarbn I agree code will be duplicated :)

**Comment by jiteshgolecha** [comment]
@alexzarbn Any updates on this? Because this is most common requirement in API development.

**Comment by michaelrimbach** [comment]
@alexzarbn any chance to get this feature added?
